NitroxClient/GameLogic/Bases/GhostMetadataApplier.cs

```diff
@@ -0,0 +1,209 @@
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata.Bases;
+using NitroxModel_Subnautica.DataStructures;
+using System.Collections;
+using UnityEngine;
+using UWE;
+
+namespace NitroxClient.GameLogic.Bases;
+
+public static class GhostMetadataApplier
+{
+    public static T From<T>(BaseGhost baseGhost) where T : GhostMetadata, new()
+    {
+        T metadata = new()
+        {
+            TargetOffset = baseGhost.targetOffset.ToDto()
+        };
+        return metadata;
+    }
+
+    public static void ApplyBasicMetadataTo(this GhostMetadata ghostMetadata, BaseGhost baseGhost)
+    {
+        baseGhost.targetOffset = ghostMetadata.TargetOffset.ToUnity();
+    }
+
+    public static GhostMetadata GetMetadataForGhost(BaseGhost baseGhost)
```

Comment:
Would split up this file. Right now it does two thins, applying metadata and retrieving it.